### PR TITLE
React Native Release 2.5.0

### PR DESCRIPTION
**Pull request by charmian01**
## Description

Bumped the versions
Updated TestApp to use latest SDKs
Updated the Changelog.md

### CODE REVIEW

File: AppCenterReactNativeShared/android/build.gradle [hunk 1]
```diff
@@ -34,8 +34,8 @@ allprojects {
         defaultConfig {
             minSdkVersion 16
             targetSdkVersion 26
-            versionCode 44
-            versionName '2.3.0'
+            versionCode 45
+            versionName '2.5.0'
```
Comment (MatkovIvan) [inline AppCenterReactNativeShared/android/build.gradle]: Why do we skip 2.4.0 here?

Comment (charmian01) [reply AppCenterReactNativeShared/android/build.gradle]: In last release, the android sdk was not updated, so I reverted all android build.gradle updates. But it seems they should not be skipped. They was updated by running wrapper script, and the wrapper script is for the npm package version, JavaScript layer.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 
 * **[Behavior change]** Fix a security issue in the way native Android crashes are processed. The exception message is now `null` and the exception stack trace is the raw Java stack trace.
 
+### App Center Data
+
+* **[Fix]** Reduced retries on Data-related operations to fail fast and avoid the perception of calls "hanging".
+
+#### iOS
+
+* **[Fix]** Fix an issue where the optional delegate method `data:didCompletePendingOperation:forDocument:withError:` would throw an exception if not implemented (when using `MSData.setRemoteOperationDelegate`).
```
Comment (dhei) [inline CHANGELOG.md]: Please remove this - `MSData.setRemoteOperationDelegate` is a native iOS sdk API, not affecting RN.iOS SDK.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
-## Version 2.5.0 (Under development)
+## Version 2.5.0
 
 ### App Center
 
 * **[Fix]** Fix dependency vulnerabilities in all App Center packages.
 
+#### iOS
+
+* **[Fix]** Fix header issues with projects not using clang modules.
```
Comment (dhei) [inline CHANGELOG.md]: I don't think this fix is relevant to RN SDK. Please remove it.

**Pull request merged**